pkg/test/nodes.go

```diff
@@ -67,18 +67,22 @@ func Node(overrides ...NodeOptions) *v1.Node {
 	}
 }
 
+// this check is done for tests that assume the nodeclaim is past registration but will not be reconciled by the NodeClaim Lifecycle controller
 func NodeClaimLinkedNode(nodeClaim *v1beta1.NodeClaim) *v1.Node {
-	return Node(
+	taints := append(nodeClaim.Spec.StartupTaints, nodeClaim.Spec.Taints...)
+	n := Node(
 		NodeOptions{
 			ObjectMeta: metav1.ObjectMeta{
 				Labels:      nodeClaim.Labels,
 				Annotations: nodeClaim.Annotations,
 				Finalizers:  nodeClaim.Finalizers,
 			},
-			Taints:      append(nodeClaim.Spec.Taints, nodeClaim.Spec.StartupTaints...),
+			Taints:      taints,
 			Capacity:    nodeClaim.Status.Capacity,
 			Allocatable: nodeClaim.Status.Allocatable,
 			ProviderID:  nodeClaim.Status.ProviderID,
 		},
 	)
+	n.Spec.Taints = append(n.Spec.Taints, v1beta1.UnregisteredNoExecuteTaint)
```

Comment:
does it matter if the unregistered taint is the last one or not? Seems that's the only reason why I'd want to have two different taint append calls here.

Reply:
Just in response to comment about reducing diff in this file from @jonathan-innis  above